frontend/src/store/modules/hacknight.js

```diff
@@ -0,0 +1,67 @@
+import axios from 'axios';
+
+export default {
+  namespaced: true,
+  state: {
+    success: null,
+    error: null
+  },
+  getters: {},
+  mutations: {
+    wasSuccess(state) {
+      state.success = true;
+    },
+    raiseError(state) {
+      state.error = true;
+    }
+  },
+  actions: {
+    createHacknight({ commit }) {
+      return new Promise((resolve, reject) => {
+        const d = new Date();
+
+        axios
+          .post('/hacknights/', {
+            date: d.toUTCString()
+          })
+          .then(
+            res => {
+              commit('wasSuccess');
+              resolve(res);
+            },
+            error => {
+              commit('raiseError');
+              reject(error);
+            }
+          );
+      });
+    },
+    getHacknight({ commit }, hacknight_id) {
+      return new Promise((resolve, reject) => {
```

Comment:
As `axios` claims to be
> Promise based HTTP client for the browser and node.js

there's probably no reason to encapsulate it inside `new Promise()` and you can simply write that down as
```js
return axios.get(`/hacknights/${id}/`).then((res) => {
  commit('wasSuccess');
  return res;
});
```
(error part is ommitted, as `axios` should raise that error by itself).